File: B-Coins.py
A = int(input())
B = int(input())
C = int(input())
X = int(input())
x = 0
ans = 0
y = 0
a = [i for i in range((A + 1)* (B + 1) * (C + 1))]
for i in range(A + 1):
    for j in range(B + 1):
        for k in range(C + 1):
            x = 500 * i + 100 * j + 50 * k
            a[y] = x
            y += 1
ans = a.count(X)
print(ans)

File: B-Coins.go
package main
import (
	"fmt"
)

func main(){
	var a, b, c, x int
	fmt.Scan(&a)
	fmt.Scan(&b)
	fmt.Scan(&c)
	fmt.Scan(&x)
	ans := 0
	for i := 0; i < a + 1; i ++ {
		for j := 0; j < b + 1; j ++ {
			y := 500 * i + 100 * j
			z := x - y
			if z / 50 >= 0 && z / 50 <= c {
				ans += 1
			}
		}
	}
	fmt.Println(ans)

}